cmd/driver-registrar/main.go

```diff
@@ -204,10 +160,35 @@ func main() {
 	// as gRPC server which replies to registration requests initiated by kubelet's
 	// pluginswatcher infrastructure. Node labeling is done by kubelet's csi code.
 	if *kubeletRegistrationPath != "" {
-		stopCh := make(chan bool)
-		k8sRegistration(csiDriverName)
-		// Block on the channel to serve registration requests from kubelet
-		<-stopCh
+		registrar := newRegistrationServer()
+		registrar.driverName = csiDriverName
+		registrar.endpoint = *kubeletRegistrationPath
+		registrar.version = supportedVersions
+		socketPath := fmt.Sprintf("/registration/%s-reg.sock", csiDriverName)
+		// Remove any socket, stale or not, but fall through for other files
+		fi, err := os.Stat(socketPath)
+		if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
+			os.Remove(socketPath)
+		}
+		// Default to only user accessible socket, caller can open up later if desired
+		oldmask := unix.Umask(0077)
+
+		glog.Infof("Starting Registration Server at: %s\n", socketPath)
+		lis, err := net.Listen("unix", socketPath)
+		if err != nil {
+			os.Exit(1)
+		}
+		unix.Umask(oldmask)
+		glog.Infof("Registration Server started at: %s\n", socketPath)
```

Comment:
nit: `glog.V(4).Infof` ?

Reply:
none of driver-registar logging uses this approach, probably in a separate PR it would make sense to refactor.